# City of Seattle Staff Age and Wage Data

In this lab we will be analyzing City of Seattle data on the age and wage of city employees. We will be asking:
1. What the mean wage is?
2. What the median wage is?
3. What the minimum wage is?
4. What the maximum wage is?
5. What the mode of the wage data is?
6. What the mean age is?
7. What the median age is?
8. What the minimum age is?
9. What the maximum age is?
10. What the mode of the aage data is?

### Data Set Preparation

In [1]:
from scipy import stats;
import math;

In [2]:
data_file = open("City_of_Seattle_Staff_Demographics.csv", "r");
staff_list = [];
for row in data_file:
    staff_list.append(row);
data_file.close();

Here I opened the file and read the file into a list of data.

In [3]:
del staff_list[0];

Here I deleted the header of the data, so that the data file only consists of data.

In [4]:
age = [];
wage = [];    
for item in range(len(staff_list)):
    #make into a list of lists of just numbers
    staff_list[item] = staff_list[item].split(",")
    
    #add data to specific lists for west and east
    age.append(int(staff_list[item][3]))
    wage.append(float(staff_list[item][4]))

print(age[:10])
print(wage[:10])

[57, 39, 52, 30, 34, 37, 65, 29, 65, 48]
[29.75, 27.18, 59.75, 38.28, 53.87, 47.98, 48.04, 33.45, 49.32, 45.734]


Here I created two lists, wage and age, and then sorted the data into the two lists (one of integers and one of floats), which I then print the first 10 numbers of to confirm whether or not the lists have the correct Data.

### Data Modeling

In [5]:
minA = min(age)
maxA = max(age)
indexMinA = age.index(minA)
indexMaxA = age.index(maxA)
print("The youngest employee is " + str(minA))
print("The oldest employee is " + str(maxA))

The youngest employee is 14
The oldest employee is 92


Here I find the minimum and maximum age before finding their indexes and then printing them.

In [6]:
age.sort();  # sort the list. It's that easy
halfwayPoint = math.floor(len(age)/2);

if (len(age) % 2) == 0: 
    medianA = (age[halfwayPoint] + age[halfwayPoint-1])/2;
else: # the halfway point is an odd number, so the median is the middle number
    medianA = age[halfwayPoint];
print("The median age is " + str(round(medianA,1)))

The median age is 45.0


Here I calculate the median age by sorting the ages and then finding the halfway point between the minimum and maximum. I then print the median.

In [7]:
totalA = 0
for a in age:
    totalA = totalA + a
averageA = totalA/len(age)
print("The mean age is " + str(round(averageA,3)))

The mean age is 44.534


Here I calculate the mean age by finding the total through a for loop before dividing it by the length and then printing it.

In [8]:
modeA = stats.mode(age)
print(modeA)

ModeResult(mode=array([48]), count=array([412]))


Finding and printing the mode of the ages by using the stats library.

In [9]:
print(wage[indexMinA])

16.0


Finding and then printing the wage made by the youngest employee.

In [10]:
minW = min(wage)
maxW = max(wage)

Finding the minimum and maximum wages. 

In [11]:
wage.sort();  # sort the list.
halfwayPoint = math.floor(len(wage)/2);
if (len(wage) % 2) == 0: 
    medianW = (wage[halfwayPoint] + wage[halfwayPoint-1])/2;
else: # the halfway point is an odd number, so the median is the middle number
    medianW = wage[halfwayPoint];
print("The median wage is " + str(round(medianW, 1)))

The median wage is 39.0


Finding the median wage by sorting the data and finding the midpoint, this is different for lists with an even length than it is with an odd length. I account for both.

In [12]:
totalW = 0.0
for w in wage:
    totalW = totalW + w
averageW = totalW/len(wage)
print("The mean wage is " + str(round(averageW,3)))

The mean wage is 40.034


I find the mean wage by adding up all of the wages and then dividing by the number of wage data points.

In [13]:
print("The Minimum wage is " + str(round(minW,2)))
print("The Maximum wage is " + str(round(maxW,2)))

The Minimum wage is 5.53
The Maximum wage is 162.84


Printing the minimum and maximum wages.

In [14]:
modeW = stats.mode(wage)
print(modeW)

ModeResult(mode=array([16.11]), count=array([423]))


Finding and then printing the mode of the data through the stats library.

### Analysis

The most common salary for a seattle city employee is close to the smallest wage. This number is only slightly higher than the city's legal minimum wage of $15.00 an hour. It is likely that the majority of these 16-year-olds are paid high school interns, working for either Seattle City Light or other city sectors

I was surprised by a fourteen-year-old being on the city's pay roll. The fourteen-year-old is being paid 16.00 dollars. This is the lowest amount a seattle city employee was paid. However, she was not the only employee making 16.00 dollars as it is the most common salary for an employee. Given age, it is unlikely that she would be employeed as a high school intern as that has a mandatory age requirement of at least 16 years of age. Looking at the data, I was able to find out that she worked at the Seattle Public Library. 

We found that the mean age was around 44.5 years with a median of 45. When the mean is this close to the median, it means there is very little skew to the data. Additionally, the minimum age is 14 with the maximum being 92, making the right tail slightly longer if it were a box and whisker plot. The mode of the data is 48 years, with 412 city employees being 48 years of age. this means that although 45 years is the median, it is not the peak. 

For wage, the mean was 40 dollars an hour, close to the median of 39. We can expect very little skew in this data. the mode however was around 16.1 dollars an hour with 423 employees making 16 dollars an hour. This mode is very close to the minimum wage of 16.0 dollars an hour. The maximum wage paid by the city, however, was 162.8 dollars an hour.

### Conclusion

City employees have an average income that is above the national average of 27.16 dollars. However, a large amount of city employees are paid below this average with the most common wage being 16.00 dollars an hour. 
The city does offer internships and jobs to highschoolers which is important especially given the resume opportunities this can present.

### Acknowledgements

Many thanks to Navya Mangipudi and Ms. Sconyers for their much appreciated help.
Additional thanks to Geeks2Geeks.com for reminding me of the round function in Python.
Additional thanks to Ms. Sconyers for providing me with code.
Additional thanks to Scott for providing me with much needed emotional encouragement.
Additional thanks to the Megan Elliot from CheatSheet for providing information on average wages in the US.